pgjdbc/src/main/java/org/postgresql/core/ParameterContext.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2020, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.core;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ParameterContext {
+  public static final ParameterContext EMPTY_CONTEXT = new ParameterContext();
+  private static final List<Integer> NO_PLACEHOLDERS = new ArrayList<>();
+  private @Nullable BindStyle bindStyle = null;
+  private @Nullable List<Integer> placeholderPositions = null;
+  private @Nullable List<String> placeholderNames = null;
+  private @Nullable List<Integer> placeholderAtPosition = null;
+
+  public static ParameterContext buildNamed(List<Integer> placeholderPositions,
+      List<String> placeholderNames) throws SQLException {
+    final ParameterContext ctx = new ParameterContext();
+    assert placeholderPositions.size() == placeholderNames.size();
```

Comment:
in a public method on a public class, it seems this should not just be asserted, but actually checked and throw IllegalArgumentException if mismatch.

Reply:
That is an oversight. But this method is only really meant for testing. So I have moved the method to a subclass in the relevant test.